internal/controllers/gateway/route_utils_test.go

```diff
@@ -0,0 +1,165 @@
+package gateway
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"k8s.io/utils/pointer"
+	gatewayv1alpha2 "sigs.k8s.io/gateway-api/apis/v1alpha2"
+)
+
+func Test_filterHostnames(t *testing.T) {
+	commonGateway := &gatewayv1alpha2.Gateway{
+		Spec: gatewayv1alpha2.GatewaySpec{
+			Listeners: []gatewayv1alpha2.Listener{
+				{
+					Name:     "listener-1",
+					Hostname: (*gatewayv1alpha2.Hostname)(pointer.StringPtr("very.specific.com")),
```

Comment:
Why not create a `func stringToGatewayHostname(h string) *gatewayv1alpha2.Hostname` that will do the conversion for us?

Seeing other invocations it seems we'd need 2 flavors: one with values and one with pointers being returned.

Reply:
good point, I'll add it